docs/development/pigs/pig-012.rst

```diff
@@ -0,0 +1,340 @@
+        fit:
+            energy_min, energy_max
+        logging:
+            level: debug
+
+    grid:
+        spatial: center, width, binsz
+        energy: min, max, nbins
+        time: min, max
+        # PSF RAD and EDISP MIGRA not exposed for now
+        # Per-obs energy_safe and roi_max ?
+    
+    observations:
+        data_store: $GAMMAPY_DATA/cta-1dc/index/gps/
+        ids: 110380, 111140, 111159
+        conesearch: 
+            ra:
+            dec:
+            radius:
+            energy_min:
+            energy_max:
+            time_min:
+            time_max:
+
+    model:
+        # Model configuration will mostly be designed in different PIG
+        sources:
+            source_1: spectrum: powerlaw, spatial: shell
+            diffuse: gal_diffuse.fits
+        background:
+            IRF
+
+**API design**
+
+The design of the high-level interface API is driven by the use cases considered, the different
+tasks (tools) identified and their responsibilities, as well as the need of a main `Analysis` 
+session object that drives and orchestrates the different tools involved, their inputs and products. 
+The `Analysis` session object will be initialized with a configuration file (see *prototype 
+configuration file* above) and will be the responsible to instatiate and run the different tools 
+classes. The tools are middle  management agents (e.g. MapMaker, ReflectedBgEstimator, ...) 
+responsible to perform the different  tasks identified in the use cases covered. `Analysis` needs
+to know how to turn configuration values into Python objects (e.g. geometries, exclusion_mask: 
+exclusion.fits to make a Map object) and also how to instatiate and run the different tools. A 
+possible solution for **dispatching** among `Analysis` and tools may be the one proposed below.
+
+.. code::
+
+    class Analysis:
+        def reduce_data(self):
+            config = self.make_data_reduction_config(self.config)
+            maker = self.make_data_recution_class(config)
+            maker.run()
+            self.datasets = maker.datasets
+
+We will need "registries" mapping options to tools classes, e.g.:
+
+.. code::
+
+    DATA_REDUCERS = {
+    "1d": "gammapy.spectrum.SpectrumExtration",
+    "3d": "gammapy.cube.MapMaker",
+    }
+
+By having registries (i.e. Python dicts), both within Gammapy and users can add their own tools, 
+in this sense the high-level interface provides also a "framework" that is extensible even at 
+runtime. For this to work, we need to have a limited number or data containers (e.g. DataSet 
+subclasses), because the tools require a certain structure of containers, modify them in-place 
+or make new containers. 
+
+**Serialisation**
+
+The user could choose in the configuration file to serialise the products delivered by the tools 
+with the help of a boolean parameter. The serialisation would then be performed as follows:
+
+    - Every tool in Gammapy knows how to serialise/read themselves
```

Comment:
In all other RST files we never indent bullet list. Please do that in this PIG also for consistency.